deployments/bridges/rialto-millau/dashboard/grafana/provisioning/dashboards/relay-rialto-to-millau-messages-dashboard.json

```diff
@@ -0,0 +1,1125 @@
+      "interval": "",
+      "legendFormat": "{{type}}",
+      "refId": "A"
+      },
+      {
+        "expr": "max_over_time(Rialto_to_Millau_MessageLane_00000000_lane_state_nonces{type=\"source_latest_generated\"}[5m]) - min_over_time(Rialto_to_Millau_MessageLane_00000000_lane_state_nonces{type=\"source_latest_generated\"}[5m])",
+        "hide": true,
+        "interval": "",
+        "legendFormat": "Messages generated in last 5 minutes",
+        "refId": "B"
+      }
+    ],
+    "thresholds": [],
+    "timeFrom": null,
+    "timeRegions": [],
+    "timeShift": null,
+    "title": "Delivery race",
+    "tooltip": {
+      "shared": true,
+      "sort": 0,
+      "value_type": "individual"
+    },
+    "type": "graph",
+    "xaxis": {
+      "buckets": null,
+      "mode": "time",
+      "name": null,
+      "show": true,
+      "values": []
+    },
+    "yaxes": [
+      {
+      "format": "short",
+      "label": null,
+      "logBase": 1,
+      "max": null,
+      "min": null,
+      "show": true
+      },
+      {
+      "format": "short",
+      "label": null,
+      "logBase": 1,
+      "max": null,
+      "min": null,
+      "show": true
+      }
+    ],
+    "yaxis": {
+      "align": false,
+      "alignLevel": null
+    }
+    },
+    {
+    "aliasColors": {},
+    "bars": false,
+    "dashLength": 10,
+    "dashes": false,
+    "datasource": "Prometheus",
+    "fieldConfig": {
+      "defaults": {
+      "custom": {}
+      },
+      "overrides": []
+    },
+    "fill": 1,
+    "fillGradient": 0,
+    "gridPos": {
+      "h": 11,
+      "w": 12,
+      "x": 12,
+      "y": 9
+    },
+    "hiddenSeries": false,
+    "id": 8,
+    "legend": {
+      "avg": false,
+      "current": false,
+      "max": false,
+      "min": false,
+      "show": true,
+      "total": false,
+      "values": false
+    },
+    "lines": true,
+    "linewidth": 1,
+    "nullPointMode": "null",
+    "options": {
+      "dataLinks": []
+    },
+    "percentage": false,
+    "pointradius": 2,
+    "points": false,
+    "renderer": "flot",
+    "seriesOverrides": [],
+    "spaceLength": 10,
+    "stack": false,
+    "steppedLine": false,
+    "targets": [
+      {
+      "expr": "label_replace(label_replace(Rialto_to_Millau_MessageLane_00000000_lane_state_nonces{type=~\"source_latest_confirmed|target_latest_received\"}, \"type\", \"Latest message confirmed by Millau to Rialto\", \"type\", \"source_latest_confirmed\"), \"type\", \"Latest message received by Rialto\", \"type\", \"target_latest_received\")",
+      "interval": "",
+      "legendFormat": "{{type}}",
+      "refId": "A"
+      }
+    ],
+    "thresholds": [],
+    "timeFrom": null,
+    "timeRegions": [],
+    "timeShift": null,
+    "title": "Confirmations race",
```

Comment:
Is there any particular reason why the Confirmation Race panel doesn't provide any alerts?

Reply:
I don't think that's the correct question - why would this (every?) panel need an alert? :) If you're talking about alerting when messages are submitted, but not being delivered, then there's separate panel with this alert - "Delivery race lags". It has graph that clearly shows when and why the alert has been triggered. On "Confirmations race" panel you may not even see the moment when the alert has been triggered - if max lag is 5 messages and nonces in last 5 minutes were in range 2000..3000, then the graph and alert scales are too different and it is not super-clear - when exactly the "lag graph" has crossed the alert line. So this panel is more about activity and overall lane state - how many messages are have been submitted, how many have been delivered.